ddl/partition.go

```diff
@@ -1212,3 +1214,43 @@ func truncateTableByReassignPartitionIDs(t *meta.Meta, tblInfo *model.TableInfo)
 	tblInfo.Partition.Definitions = newDefs
 	return nil
 }
+
+func onAlterTablePartition(t *meta.Meta, job *model.Job) (int64, error) {
+	var partitionID int64
+	var rules []*placement.Rule
+	err := job.DecodeArgs(&partitionID, &rules)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return 0, errors.Trace(err)
+	}
+
+	ver, err := t.GetSchemaVersion()
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+
+	tblInfo, err := getTableInfoAndCancelFaultJob(t, job, job.SchemaID)
+	if err != nil {
+		return ver, err
+	}
+
+	ptInfo := tblInfo.GetPartitionInfo()
+	if ptInfo.GetNameByID(partitionID) == "" {
+		job.State = model.JobStateCancelled
+		return ver, errors.Trace(table.ErrUnknownPartition.GenWithStackByArgs("drop?", tblInfo.Name.O))
```

Comment:
`drop?` is a typo?

Reply:
Nope, it is intended. Partition may have been dropped. I found a similar error at https://github.com/pingcap/tidb/blob/c7c98d738068931335b586f7adfbeb184cf0ab70/ddl/partition.go#L937, so i wrote like that.

Reply:
Got it.